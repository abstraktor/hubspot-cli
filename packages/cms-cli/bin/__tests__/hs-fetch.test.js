/**
 * @jest-environment ./CommandTestEnvironment.js
 */

const path = require('path');
const { node: execNode } = require('execa');

const remoteRoot = 'cms-theme-boilerplate';

const { hsPath, hscmsPath, localDir } = global;

const testEach = test.each`
  exe        | script
  ${'hs'}    | ${hsPath}
  ${'hscms'} | ${hscmsPath}
`;

testEach('$exe fetch --help', async ({ script }) => {
  const { stdout } = await execNode(script, ['fetch', '--help']);
  expect(stdout).toMatchSnapshot();
});

testEach('$exe fetch (missing src)', async ({ script }) => {
  let stderr;
  try {
    await execNode(script, ['fetch']);
  } catch (err) {
    ({ stderr } = err);
  }
  expect(stderr).toBeTruthy();
});

testEach('$exe fetch --portal=invalid', async ({ script }) => {
  let stderr;
  try {
    await execNode(script, ['fetch', '--portal=invalid']);
  } catch (err) {
    ({ stderr } = err);
  }
  expect(stderr).toBeTruthy();
});

testEach(`$exe fetch ${remoteRoot}/js`, async ({ script }) => {
  const src = `${remoteRoot}/js`;
  const dest = path.join(localDir, 'foo');
  const { stdout } = await execNode(script, ['fetch', src, dest]);
  expect(stdout).toBeTruthy();
});

testEach('$exe fetch /does-not-exist', async ({ script }) => {
  const src = '/does-not-exist';
  const dest = localDir;
  const { stderr } = await execNode(script, ['fetch', src, dest]);
  expect(stderr).toBeTruthy();
});
