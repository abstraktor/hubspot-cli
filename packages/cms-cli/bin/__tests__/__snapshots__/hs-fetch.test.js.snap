// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hs fetch --help 1`] = `
"Usage: hs fetch [options]

Fetch a file, directory or module from HubSpot and write to a path on your computer

Options:
  -V, --version      output the version number
  --config <config>  path to a config file
  --portal <portal>  HubSpot portal id or name from config
  --debug            set log level to debug
  --overwrite        overwrite existing files
  --mode <mode>      read from <draft | publish>
  -h, --help         output usage information"
`;

exports[`hscms fetch --help 1`] = `
"Usage: hscms fetch [options]

Fetch a file, directory or module from HubSpot and write to a path on your computer

Options:
  -V, --version      output the version number
  --config <config>  path to a config file
  --portal <portal>  HubSpot portal id or name from config
  --debug            set log level to debug
  --overwrite        overwrite existing files
  --mode <mode>      read from <draft | publish>
  -h, --help         output usage information"
`;
